{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Lab 4\n",
    "## Arc I\n",
    "### Mohsen Ahmadkhani"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'geopandas'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-1-972ed97201ec>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0murllib\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparse\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0murlencode\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mpandas\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 5\u001b[1;33m \u001b[1;32mimport\u001b[0m \u001b[0mgeopandas\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mgpd\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      6\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0mshapely\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mgeometry\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mPoint\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mmatplotlib\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpyplot\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mplt\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'geopandas'"
     ]
    }
   ],
   "source": [
    "# Download temperature data from NDAWN\n",
    "import requests\n",
    "from urllib.parse import urlencode\n",
    "import pandas as pd \n",
    "import geopandas as gpd\n",
    "from shapely.geometry import Point\n",
    "import matplotlib.pyplot as plt\n",
    "import folium\n",
    "import time\n",
    "\n",
    "def get_weather(start_date, end_date, station_count = None):\n",
    "    df = []\n",
    "    row = 1\n",
    "    for station in range(1, station_count):\n",
    "        base_url = 'https://ndawn.ndsu.nodak.edu/table.csv?'\n",
    "        params = f\"station={str(station)}&begin_date={start_date}&end_date={end_date}&ttype=daily&quick_pick=&variable=ddmxt&variable=ddmnt&variable=ddavt&variable=ddbst&variable=ddtst&variable=ddws&variable=ddmxws&variable=ddsr&variable=ddr&variable=dddp&variable=ddwc\"\n",
    "        url = base_url+params\n",
    "        r = requests.get(url)\n",
    "\n",
    "        with open(\"response.csv\", \"w\") as text_file:\n",
    "            text_file.write(r.content.decode('utf-8'))\n",
    "\n",
    "        if row == 1:\n",
    "            instance = pd.read_csv('response.csv', header=[0,1],skiprows=3)\n",
    "            if instance.iloc[0][0].find('<') == -1: \n",
    "                df = instance\n",
    "                row = 0\n",
    "        else:\n",
    "            instance = pd.read_csv('response.csv', header=[0,1],skiprows=3)\n",
    "            df = pd.concat([df, instance], ignore_index=True)\n",
    "    \n",
    "    return df\n",
    "            \n",
    "\n",
    "\n",
    "today_date = list(time.localtime())\n",
    "yesterday = str(today_date[0]) + '-' + str(today_date[1]) + '-' + str(today_date[2]-1)\n",
    "last_month = str(today_date[0]) + '-' + str(today_date[1]-1) + '-' + str(today_date[2]-1)\n",
    "\n",
    "start_date = last_month\n",
    "end_date = yesterday\n",
    "\n",
    "df = get_weather(start_date, end_date, 300) # get the data for all ND weather stations\n",
    "df.columns = [f'{i}_{j}' for i, j in df.columns] # concat two header rows into one \n",
    "\n",
    "# df.to_csv('weather_monthly.csv', encoding='utf-8', index=False)\n",
    "\n",
    "stations_gdf = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.Longitude_deg, df.Latitude_deg))\n",
    "stations_gdf.crs = 4326\n",
    "\n",
    "stations_gdf.head()\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Collecting geopandas\n",
      "  Using cached geopandas-0.9.0-py2.py3-none-any.whl (994 kB)\n",
      "Requirement already satisfied: pandas>=0.24.0 in c:\\program files\\arcgis\\pro\\bin\\python\\envs\\arcgispro-py3\\lib\\site-packages (from geopandas) (1.0.3)\n",
      "Collecting fiona>=1.8\n",
      "  Using cached Fiona-1.8.20.tar.gz (1.3 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.19.tar.gz (1.3 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.18.tar.gz (1.3 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.17.tar.gz (1.3 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.16.tar.gz (1.3 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.15.tar.gz (1.3 MB)"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_b4c62dc81dc54a44a8cccc07e3ca16e7\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_b4c62dc81dc54a44a8cccc07e3ca16e7\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-dgoor45_'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_b4c62dc81dc54a44a8cccc07e3ca16e7\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/ec/f7/093890341a7e8fbfcdfa04caf4dfb588ebab32c13ceaa6a3819da79ea106/Fiona-1.8.20.tar.gz#sha256=a70502d2857b82f749c09cb0dea3726787747933a2a1599b5ab787d74e3c143b (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_402bc121533e456f85d0d928fdf0d282\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_402bc121533e456f85d0d928fdf0d282\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-c1m0vmr_'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_402bc121533e456f85d0d928fdf0d282\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/a0/d9/6042aeb073d11341f7726de0586ff71c13117c34959dcf07bd4ee6d4b93e/Fiona-1.8.19.tar.gz#sha256=b9059e0b29c2e9e6b817e53f941e77e1aca7075f986005d38db307067b60458f (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_d27926b571de43bba391a9b54ad8aff9\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_d27926b571de43bba391a9b54ad8aff9\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-4k6z2c8_'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_d27926b571de43bba391a9b54ad8aff9\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/9f/e8/401cdaa58d862a25c4b3365acf7d2bd7ac77191e3dc9acdcdac0eff20ff0/Fiona-1.8.18.tar.gz#sha256=b732ece0ff8886a29c439723a3e1fc382718804bb057519d537a81308854967a (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_403b413d0b794a74ae0442c26b9288ab\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_403b413d0b794a74ae0442c26b9288ab\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-a8xd3_dc'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_403b413d0b794a74ae0442c26b9288ab\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/88/62/69347ba2c41b526e1953c4cb66d51170b2869808863c03af202ba0121670/Fiona-1.8.17.tar.gz#sha256=716201c21246587f374785bec6d6a20a984fe1f6c2b0e83bf15127eb8f724d0c (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_706af5e181fc47499eafb5136450b326\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_706af5e181fc47499eafb5136450b326\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-kc4a6anj'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_706af5e181fc47499eafb5136450b326\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/1e/60/dfc6115a11338d8aa96cacd8c60635223d9c97d61d556c90acc5dfd663fa/Fiona-1.8.16.tar.gz#sha256=fd6dfb65959becc916e9f6928618bfd59c16cdbc413ece0fbac61489cd11255f (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_b90d3c86cc934a3d86a77b6e323936e4\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_b90d3c86cc934a3d86a77b6e323936e4\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-mbpczdzc'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_b90d3c86cc934a3d86a77b6e323936e4\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/55/2f/17450ec2c8fcc720a8a3e4d9a383499508475c7cfb90f7eca9fb585ac598/Fiona-1.8.15.tar.gz#sha256=3b1c9b5c834fae2fe947cfaea176db890bc6750d1a6ba9f54d969c19ffcd191e (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_943d2df3523946c0a2d9dd485c58588b\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_943d2df3523946c0a2d9dd485c58588b\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-b5aq67o9'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_943d2df3523946c0a2d9dd485c58588b\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/5c/fd/5ec54f2d9b3d5dd23dd443fad5630d6b872e2664814c68b856c47e0d65af/Fiona-1.8.14.tar.gz#sha256=6eac038206c89d2cf5f99ea38b81cc228dc21eac5f47870a9a32d453b0007f4d (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_c186a475fa1947d395a28505677487ea\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_c186a475fa1947d395a28505677487ea\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-f7s6pod9'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_c186a475fa1947d395a28505677487ea\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/6d/42/f4a7cac53b28fa70e9a93d0e89a24d33e14826dad6644b699362ad84dde0/Fiona-1.8.13.post1.tar.gz#sha256=1a432bf9fd56f089256c010da009c90d4a795c531a848132c965052185336600 (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_65f125c75ef5467891fbe7f65a684002\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_65f125c75ef5467891fbe7f65a684002\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-9ln9kuey'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_65f125c75ef5467891fbe7f65a684002\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/be/04/31d0a6f03943b1684f32c9b861be40c1fd282468fa6bd54ddf4a774e6b0f/Fiona-1.8.13.tar.gz#sha256=5ec34898c8b983a723fb4e949dd3e0ed7e691c303e51f6bfd61e52ac9ac813ae (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_0c8d9ac6c6d44f60bfe8cf82cfe449ae\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_0c8d9ac6c6d44f60bfe8cf82cfe449ae\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-fcmdvpte'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_0c8d9ac6c6d44f60bfe8cf82cfe449ae\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/97/d8/feab39987296437fbdc3029fb39752a14355d217d73b93471010b8dd63a3/Fiona-1.8.12.tar.gz#sha256=c9266ddf6ae2a64fcea20014ddf27f800ac07584f2fdb09c2a02f3b3a52e371c (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_d9770dd6f69041f7983be15b9d540eb9\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_d9770dd6f69041f7983be15b9d540eb9\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-6rk4pkcv'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_d9770dd6f69041f7983be15b9d540eb9\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/9d/f4/0a0ddc6174c4a93679b5f1dd3535e7ef8989828e6d5f86112de681f8c87b/Fiona-1.8.11.tar.gz#sha256=1e7ca9e051f5bffa1c43c70d573da9ca223fc076b84fa73380614fc02b9eb7f6 (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_aae6087fea7b4ff0bc29afc1123aa597\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_aae6087fea7b4ff0bc29afc1123aa597\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-wshed2nv'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_aae6087fea7b4ff0bc29afc1123aa597\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/13/73/f80b491ed8559326fab202a6d6333a3cd6e8be1e9d782bc6c0d03d476457/Fiona-1.8.10.tar.gz#sha256=ff562eb2f3960e21f8c7f050ddd7f47a763869ea14afc2234a40df72666c6a2c (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_10b0069f10eb423ea9fd389671a67b8e\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_10b0069f10eb423ea9fd389671a67b8e\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-lmdogeic'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_10b0069f10eb423ea9fd389671a67b8e\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/9b/52/45e75507660ce0e86176d0f59b659560f687e2c7e9ebf82a10e7dcd2d3b7/Fiona-1.8.9.post2.tar.gz#sha256=210fb038b579fab38f35ddbdd31b9725f4d5099b3edfd4b87c983e5d47b79983 (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_a9423451ef7c405aa302df30d467c2f7\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_a9423451ef7c405aa302df30d467c2f7\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-ubo_m5zh'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_a9423451ef7c405aa302df30d467c2f7\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/0e/a4/d9dd7399be809d3990f5000fb6ae43189ea26ae88be1bed3a4c9ddc1becc/Fiona-1.8.9.post1.tar.gz#sha256=d5e0ea0b8addffe9cba4cb59e2bd495b015230e7a1b1597974f5048211930199 (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_58b6cc4ae96845ea874d3bac60cfaeeb\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_58b6cc4ae96845ea874d3bac60cfaeeb\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-hq3zzof6'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_58b6cc4ae96845ea874d3bac60cfaeeb\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/ad/92/dcbd8c54d697c22f299b5af63b6df3acfbd06c6d72e249614c05be99337c/Fiona-1.8.9.tar.gz#sha256=4dd6e2f5327c1174143c7c8594a75d373bc72f2c9a2a6daee312c3186a128add (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_8769fdcda7874676b64a60f1ed628d78\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_8769fdcda7874676b64a60f1ed628d78\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-7xdm9_0w'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_8769fdcda7874676b64a60f1ed628d78\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/94/7f/e288db1ad63d759d494c30caae34f865e0c6927588c490705e91b7134193/Fiona-1.8.8.tar.gz#sha256=711c3be73203b37812992089445a1e4e9d3d6b64e667389f7b15406e15a91e83 (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_723d6ae6791d4d4fab93a53dfb67e2e6\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_723d6ae6791d4d4fab93a53dfb67e2e6\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-a2szi_id'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_723d6ae6791d4d4fab93a53dfb67e2e6\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/78/62/daafd070aebefa639df247705b97f13f7cfad43895581b5cae41bd886709/Fiona-1.8.7.tar.gz#sha256=a55a23615bad3e142d4e4cda97bb5de83c778a80049222e9dffae93c13b5cf93 (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_17ff7b4d7503473f97444a69daff2b09\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_17ff7b4d7503473f97444a69daff2b09\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-u38i_g8h'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_17ff7b4d7503473f97444a69daff2b09\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/41/9d/63696e7b1de42aad294d4781199a408bec593d8fdb80a2b4a788c911a33b/Fiona-1.8.6.tar.gz#sha256=fa31dfe8855b9cd0b128b47a4df558f1b8eda90d2181bff1dd9854e5556efb3e (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_eadf3347712841819af545bb6034075e\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_eadf3347712841819af545bb6034075e\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-3_h73x0f'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_eadf3347712841819af545bb6034075e\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/7b/af/1c2c83c4a8363a4ce9fea817b1910b5e071bed012e18257faa2a0ab3cfe7/Fiona-1.8.5.tar.gz#sha256=4f5cc2d449edbbf693c83e24cdada72de7c41297383d16fcc92387eb445e9d35 (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_e093990ea84f4fdbb21703dbe12df891\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_e093990ea84f4fdbb21703dbe12df891\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-36ncd42_'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_e093990ea84f4fdbb21703dbe12df891\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/3a/16/84960540e9fce61d767fd2f0f1d95f4c63e99ab5d8fddc308e8b51b059b8/Fiona-1.8.4.tar.gz#sha256=aec9ab2e3513c9503ec123b1a8573bee55fc6a66e2ac07088c3376bf6738a424 (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_5ac2c5ccb4464b7c9906d76a96208998\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_5ac2c5ccb4464b7c9906d76a96208998\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-_2wkpy8e'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_5ac2c5ccb4464b7c9906d76a96208998\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/46/d1/fcdb32513a03abfde0d97fd9782ce0f8cc0540fa6c6ce783e87b94064964/Fiona-1.8.3.tar.gz#sha256=3e831100a23c3b6cd32b98baf0c9e2119d909b44a5cf4533d3625f61dcf2d2b1 (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_c48b1256915f4e98961d58822adf96e1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_c48b1256915f4e98961d58822adf96e1\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-tzm7_2d4'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_c48b1256915f4e98961d58822adf96e1\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/25/50/0466d5d83e1859c5ca38351ee932d64cc5635f9d4dad522879e58f4b0018/Fiona-1.8.2.tar.gz#sha256=4c6419b7ac29136708029f6a44b4ccd458735a4d241016c7b1bab41685c08d8f (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_4c8569e90a914b20b3a28e9c7dc712d0\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_4c8569e90a914b20b3a28e9c7dc712d0\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-e6148o68'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_4c8569e90a914b20b3a28e9c7dc712d0\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/3e/5f/0c6704efeea2ff3fba7f54cc6ec38070157f21bc1cffa7bdfa7c9f6b8f7a/Fiona-1.8.1.tar.gz#sha256=4c34bb4c5cd788aaf14e5484c3b7de407b1a8a7c7b2d29bbb2e8b37931e83b8d (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_c2a851150969455e8ea4f795d47cbc27\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_c2a851150969455e8ea4f795d47cbc27\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-pip-egg-info-l870orxz'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_c2a851150969455e8ea4f795d47cbc27\\\n",
      "  Complete output (1 lines):\n",
      "  A GDAL API version must be specified. Provide a path to gdal-config using a GDAL_CONFIG environment variable or use a GDAL_VERSION environment variable.\n",
      "  ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/2a/bd/c1efc2680f338e5941121c776d6323af6b9698ac739e22ba523cee348a7f/Fiona-1.8.0.tar.gz#sha256=20141a9ece06daa7bb4333fba640c2fe39a49f8aca5492d1da8595d41e91844a (from https://pypi.org/simple/fiona/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_ec4fd6b999ab4eb9946261e2c5a51325\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_ec4fd6b999ab4eb9946261e2c5a51325\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-wheel-lv18192c'\n",
      "       cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_ec4fd6b999ab4eb9946261e2c5a51325\\\n",
      "  Complete output (42 lines):\n",
      "  C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\lib\\distutils\\dist.py:261: UserWarning: Unknown distribution option: 'metadata_version'\n",
      "    warnings.warn(msg)\n",
      "  C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\lib\\distutils\\dist.py:261: UserWarning: Unknown distribution option: 'requires_python'\n",
      "    warnings.warn(msg)\n",
      "  C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\lib\\distutils\\dist.py:261: UserWarning: Unknown distribution option: 'requires_external'\n",
      "    warnings.warn(msg)\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running build_py\n",
      "  creating build\n",
      "  creating build\\lib.win-amd64-3.6\n",
      "  creating build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\collection.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\compat.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\crs.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\drvsupport.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\errors.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\inspector.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\rfc3339.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\tool.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\transform.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  copying .\\fiona\\__init__.py -> build\\lib.win-amd64-3.6\\fiona\n",
      "  creating build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\bounds.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\calc.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\cat.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\collect.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\distrib.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\dump.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\env.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\filter.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\helpers.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\info.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\insp.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\load.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\ls.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\main.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\options.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  copying .\\fiona\\fio\\__init__.py -> build\\lib.win-amd64-3.6\\fiona\\fio\n",
      "  running build_ext\n",
      "  building 'fiona._transform' extension\n",
      "  error: Microsoft Visual C++ 14.0 is required. Get it with \"Build Tools for Visual Studio\": https://visualstudio.microsoft.com/downloads/\n",
      "  ----------------------------------------\n",
      "  ERROR: Failed building wheel for fiona\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_ec4fd6b999ab4eb9946261e2c5a51325\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_ec4fd6b999ab4eb9946261e2c5a51325\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-record-qc9oxcfz\\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\\Users\\umn-ahmad178\\AppData\\Roaming\\Python\\Python36\\Include\\fiona'\n",
      "         cwd: C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-install-pam_k3he\\fiona_ec4fd6b999ab4eb9946261e2c5a51325\\\n",
      "    Complete output (12 lines):\n",
      "    C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\lib\\distutils\\dist.py:261: UserWarning: Unknown distribution option: 'metadata_version'\n",
      "      warnings.warn(msg)\n",
      "    C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\lib\\distutils\\dist.py:261: UserWarning: Unknown distribution option: 'requires_python'\n",
      "      warnings.warn(msg)\n",
      "    C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\lib\\distutils\\dist.py:261: UserWarning: Unknown distribution option: 'requires_external'\n",
      "      warnings.warn(msg)\n",
      "    running install\n",
      "    running build\n",
      "    running build_py\n",
      "    running build_ext\n",
      "    building 'fiona._transform' extension\n",
      "    error: Microsoft Visual C++ 14.0 is required. Get it with \"Build Tools for Visual Studio\": https://visualstudio.microsoft.com/downloads/\n",
      "    ----------------------------------------\n",
      "ERROR: Command errored out with exit status 1: 'C:\\Program Files\\ArcGIS\\Pro\\bin\\Python\\envs\\arcgispro-py3\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_ec4fd6b999ab4eb9946261e2c5a51325\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\UMN-AH~1\\\\AppData\\\\Local\\\\Temp\\\\pip-install-pam_k3he\\\\fiona_ec4fd6b999ab4eb9946261e2c5a51325\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\UMN-AH~1\\AppData\\Local\\Temp\\pip-record-qc9oxcfz\\install-record.txt' --single-version-externally-managed --user --prefix= --compile --install-headers 'C:\\Users\\umn-ahmad178\\AppData\\Roaming\\Python\\Python36\\Include\\fiona' Check the logs for full command output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.14.tar.gz (1.3 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.13.post1.tar.gz (1.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.13.tar.gz (1.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.12.tar.gz (1.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.11.tar.gz (1.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.10.tar.gz (1.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.9.post2.tar.gz (1.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.9.post1.tar.gz (1.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.9.tar.gz (1.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.8.tar.gz (1.7 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.7.tar.gz (1.7 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.6.tar.gz (1.7 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.5.tar.gz (1.7 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.4.tar.gz (1.1 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.3.tar.gz (1.1 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.2.tar.gz (1.2 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.1.tar.gz (1.1 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "  Using cached Fiona-1.8.0.tar.gz (1.4 MB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n",
      "Collecting geopandas\n",
      "  Using cached geopandas-0.8.2-py2.py3-none-any.whl (962 kB)\n",
      "Collecting fiona\n",
      "  Using cached Fiona-1.7.13.tar.gz (731 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Requirement already satisfied: shapely in c:\\users\\umn-ahmad178\\appdata\\roaming\\python\\python36\\site-packages (from geopandas) (1.7.1)\n",
      "Requirement already satisfied: pyproj>=2.2.0 in c:\\users\\umn-ahmad178\\appdata\\roaming\\python\\python36\\site-packages (from geopandas) (3.0.1)\n",
      "Requirement already satisfied: pytz>=2017.2 in c:\\program files\\arcgis\\pro\\bin\\python\\envs\\arcgispro-py3\\lib\\site-packages\\pytz-2019.3-py3.6.egg (from pandas>=0.24.0->geopandas) (2019.3)\n",
      "Requirement already satisfied: numpy>=1.13.3 in c:\\program files\\arcgis\\pro\\bin\\python\\envs\\arcgispro-py3\\lib\\site-packages (from pandas>=0.24.0->geopandas) (1.18.1)\n",
      "Requirement already satisfied: python-dateutil>=2.6.1 in c:\\program files\\arcgis\\pro\\bin\\python\\envs\\arcgispro-py3\\lib\\site-packages (from pandas>=0.24.0->geopandas) (2.8.1)\n",
      "Requirement already satisfied: certifi in c:\\program files\\arcgis\\pro\\bin\\python\\envs\\arcgispro-py3\\lib\\site-packages (from pyproj>=2.2.0->geopandas) (2020.4.5.1)\n",
      "Requirement already satisfied: cligj>=0.4 in c:\\users\\umn-ahmad178\\appdata\\roaming\\python\\python36\\site-packages (from fiona->geopandas) (0.7.2)\n",
      "Requirement already satisfied: click-plugins in c:\\users\\umn-ahmad178\\appdata\\roaming\\python\\python36\\site-packages (from fiona->geopandas) (1.1.1)\n",
      "Requirement already satisfied: six>=1.7 in c:\\program files\\arcgis\\pro\\bin\\python\\envs\\arcgispro-py3\\lib\\site-packages (from fiona->geopandas) (1.14.0)\n",
      "Requirement already satisfied: munch in c:\\users\\umn-ahmad178\\appdata\\roaming\\python\\python36\\site-packages (from fiona->geopandas) (2.5.0)\n",
      "Requirement already satisfied: click>=4.0 in c:\\users\\umn-ahmad178\\appdata\\roaming\\python\\python36\\site-packages (from cligj>=0.4->fiona->geopandas) (8.0.3)\n",
      "Requirement already satisfied: colorama in c:\\program files\\arcgis\\pro\\bin\\python\\envs\\arcgispro-py3\\lib\\site-packages (from click>=4.0->cligj>=0.4->fiona->geopandas) (0.4.3)\n",
      "Requirement already satisfied: importlib-metadata in c:\\program files\\arcgis\\pro\\bin\\python\\envs\\arcgispro-py3\\lib\\site-packages (from click>=4.0->cligj>=0.4->fiona->geopandas) (1.5.0)\n",
      "Requirement already satisfied: zipp>=0.5 in c:\\program files\\arcgis\\pro\\bin\\python\\envs\\arcgispro-py3\\lib\\site-packages (from importlib-metadata->click>=4.0->cligj>=0.4->fiona->geopandas) (3.1.0)\n",
      "Building wheels for collected packages: fiona\n",
      "  Building wheel for fiona (setup.py): started\n",
      "  Building wheel for fiona (setup.py): finished with status 'error'\n",
      "  Running setup.py clean for fiona\n",
      "Failed to build fiona\n",
      "Installing collected packages: fiona, geopandas\n",
      "    Running setup.py install for fiona: started\n",
      "    Running setup.py install for fiona: finished with status 'error'\n"
     ]
    }
   ],
   "source": [
    "!pip install geopandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#aggregate by averaging the average daily temperature to get to the monthly average\n",
    "stations_gdf = stations_gdf.groupby('Station Name_Unnamed: 0_level_1').mean() \n",
    "stations_gdf = gpd.GeoDataFrame(stations_gdf, geometry=gpd.points_from_xy(stations_gdf.Longitude_deg, stations_gdf.Latitude_deg))\n",
    "stations_gdf.crs = 4326\n",
    "stations_gdf = stations_gdf.reset_index()\n",
    "stations_gdf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot an interactive map\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore') \n",
    "\n",
    "m = folium.Map(location = [44.9779006, -93.1831530], tiles='OpenStreetMap' , zoom_start = 12) # tiles=\"Stamen Toner\"\n",
    "\n",
    "for _, r in stations_gdf.iterrows():\n",
    "\n",
    "    sim_geo = gpd.GeoSeries(r['geometry']) \n",
    "    geo_j = sim_geo.to_json()\n",
    "   \n",
    "    geo_j = folium.GeoJson(data=geo_j, \n",
    "                           style_function = lambda x: {'color': 'red', 'weight': 1,  'fillColor': 'YlGnBu'})\n",
    "    folium.Popup(f\"<i>Station: {r['Station Name_Unnamed: 0_level_1']}, Average Monthly Temperature: {r['Avg Temp_Degrees F']} °F</i>\").add_to(geo_j)\n",
    "    folium.Tooltip(f\"<i>Station: {r['Station Name_Unnamed: 0_level_1']} <br> Avg Monthly Temperature: {r['Avg Temp_Degrees F']} °F</i>\").add_to(geo_j)\n",
    "    geo_j.add_to(m)\n",
    "\n",
    "\n",
    "sw = [44.867066, -104.248]\n",
    "ne = [48.884, -93.065245]\n",
    "\n",
    "m.fit_bounds([sw, ne]) \n",
    "\n",
    "m\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = geopandas.GeoDataFrame(stations_gdf, geometry='geom')\n",
    "df.crs= \"+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs\"\n",
    "df.to_file('ndawn_weather.shp', driver='ESRI Shapefile')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import arcpy\n",
    "# dem_3m = r'C:\\Users\\umn-ahmad178\\Desktop\\arc1lab2\\elevation_data.gdb\\dem_3m_m'\n",
    "# p = r'C:\\Users\\umn-ahmad178\\Desktop\\arc1lab2\\elevation_data.gdb'\n",
    "p = r'C:\\Users\\umn-ahmad178\\Desktop\\lab4arc1'\n",
    "arcpy.env.workspace = p  # setting workspace to the current directory\n",
    "arcpy.env.overwriteOutput = True\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ndawn_weather.shp']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arcpy.ListFeatureClasses()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "Idw(in_point_features, z_field, {cell_size}, {power}, {search_radius}, {in_barrier_polyline_features})\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#IDW\n",
    "from arcpy.sa import *\n",
    "\n",
    "# Set local variables\n",
    "inPointFeatures = 'ndawn_weather.shp'\n",
    "zField = \"Avg Temp F\"\n",
    "cellSize = 0.014477764\n",
    "power = 2\n",
    "searchRadius = RadiusVariable(10, 150000)\n",
    "\n",
    "# Execute IDW\n",
    "outIDW = Idw(inPointFeatures, zField, cellSize, power, searchRadius)\n",
    "\n",
    "# Save the output \n",
    "\n",
    "outIDW.save(r'C:\\Users\\umn-ahmad178\\Desktop\\lab4arc1\\idwout.tif')\n",
    "outKrig.save(r\"C:\\Users\\umn-ahmad178\\Desktop\\lab4arc1\\krigout.tif\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Universal Kriging\n",
    "outKrig = Kriging(inPointFeatures, zField, KrigingModelOrdinary(\"Linear drift\", 0.014477764, 2.6, 542, 0), 2000, RadiusFixed(20000, 1))\n",
    "outKrig.save(r\"C:\\Users\\umn-ahmad178\\Desktop\\lab4arc1\\uni_krigout.tif\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Ordinary Kriging\n",
    "outKrig = Kriging(inPointFeatures, zField, KrigingModelOrdinary(\"GAUSSIAN\", 0.014477764, 2.6, 542, 0), 2000, RadiusFixed(20000, 1))\n",
    "outKrig.save(r\"C:\\Users\\umn-ahmad178\\Desktop\\lab4arc1\\krigout_ordinary.tif\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "arcpy.LocalPolynomialInterpolation_ga(\n",
    "    inPointFeatures, zField, \"outLPI\", r\"C:\\Users\\umn-ahmad178\\Desktop\\lab4arc1\\idwout.tif\\polynomial.tif\", \"2000\",\n",
    "    \"2\", arcpy.SearchNeighborhoodSmooth(300000, 300000, 0, 0.5), \"QUARTIC\", \n",
    "    \"\", \"\", \"\", \"\", \"PREDICTION\")\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
